documentation/CAMARA-API-access-and-user-consent.md

```diff
@@ -287,6 +288,94 @@ If some use case(s) for an API point to "Off-net" scenarios and where Consumptio
+}
+```
+
+Example Token Request:
+
+```
+POST /token.oauth2 HTTP/1.1
+Host: as.example.com
+Content-Type: application/x-www-form-urlencoded
+
+grant_type=urn%3Aietf%3Aparams%3Aoauth%3Agrant-type%3Ajwt-bearer
+&assertion=eyJhbGciOiJFUzI1NiIsImtpZCI6IjE2In0.
+eyJpc3Mi[...omitted for brevity...].
+J9l-ZhwP[...omitted for brevity...]
+```
+
++ The Authorization Server MUST validate the signatur of the signed assertion.
++ The Authorization Server MUST validate that the value of "iss" is associated with the signer of the assertion.
++ The Authorization Server MUST validate the "aud" value.
++ The Authorization Server MUST validate "exp" and "iat" values and the lifetime of the assertion MUST not exceed 300 seconds.
++ The Authorization Server MUST validate that the Purpose is a "legitimate-use" purpose e.g. dpv:FraudPreventionAndDetection
++ The Authorization Server MUST validate that the client is allowed to use the Purpose.
++ The Authorization Server MUST validate that the client is allowed to use the scopes.
++ The Authorization Server MUST validate the subject. In this example the TS.43 token is send to the API Provider's  Entitlement Server.
+
+```mermaid
+sequenceDiagram
+
+title NumberVerification with TS.43 Temporary Token
+
+participant API Consumer
+
+participant API Consumer Backend
+
+box Carrier
+participant Authorization Server
+participant CAMARA API
+participant Entitlement Server
+end
+
+
+note over API Consumer, Entitlement Server: Mobile App gets subscriber from Entitlement Server<br/>through Mobile Operating System
+API Consumer ->> API Consumer Backend: POST phonenumber, temporaryToken
+
+note over API Consumer Backend, Authorization Server: {Camara JWT Bearer Flow}<br/>sub=operatortoken:<temporaryToken><br/>scope number-verification<br/>Purpose legitimate use 
+
+API Consumer Backend ->> Authorization Server: POST signed assertion<br/>assertion=<br/> {scope=dpv:FraudPreventionAndDetection number-verification:device-phone-number:read,<br/>  sub=operatortoken:<temporaryToken>, ...}
+
+note over Authorization Server, Authorization Server: Validate Request
+
+Authorization Server ->> Authorization Server: create ASAC Eligibility Token
+Authorization Server ->> Entitlement Server: {TS.43} GetPhoneNumber<br/>(tempToken, eligibilityToken)
+
+Entitlement Server ->> Entitlement Server: Validate tempToken
+
+Entitlement Server ->> Authorization Server: 200 OK MSISDN
+
+Authorization Server ->> Authorization Server: check consent based on<br/>client_id, scopes, purpose, MSISDN
+
+Authorization Server ->> Authorization Server: create CAMARA access token
+
+Authorization Server ->> API Consumer Backend: return access token
+
+API Consumer Backend ->> CAMARA API: CAMARA NumberVerification API /verify <br/>Authorization: Bearer <acessToken><br/>phoneNumber=phonenumber
+
+CAMARA API ->> API Consumer Backend: devicePhoneNumberVerified (boolean)
+
+API Consumer Backend ->> API Consumer: devicePhoneNumberVerified
+```
+
```

Comment:
* In other flows we add some flow description, it would be advisable to add this as well.
* In other flows, we add a technical rule set to explain the applicability of the flow.